src/dataset.cpp

```diff
@@ -247,10 +247,10 @@ const char *DatasetInfo::getCastDatatype() const
 
 void DatasetInfo::printInfo(std::string dataset_type) const
 {
-    printf("%s dataset: %s, resolution=%lld",
+    printf("%s dataset: %s, resolution=%lu",
```

Comment:
The correct way in C99 to print `size_t` is `%zu`, but if you want to be compatible with all compilers, better cast the argument to long long and use `%lld`.

Reply:
You can't print `-1` as an unsigned number.